src/main/java/filter/expression/Qualifier.java

```diff
@@ -717,6 +714,61 @@ private boolean milestoneSatisfies(IModel model, TurboIssue issue) {
         String contents = content.get().toLowerCase();
         String title = milestone.get().getTitle().toLowerCase();
 
+        List<TurboMilestone> allMilestones = model.getMilestones();
+        Collections.sort(allMilestones, new TurboMilestoneDueDateComparator());
+
+        int currentIndex = -1;
+
+        for (TurboMilestone checker : allMilestones) {
+            // current milestone must be open
+            if (checker.isOpen()) {
+                // if milestone is overdue but have no open issue, ignore
+                if (checker.getDueDate().isPresent() &&
+                        checker.getDueDate().get().isBefore(LocalDate.now()) &&
+                        checker.getOpenIssues() == 0) {
+                    continue;
+                }
+                currentIndex = allMilestones.indexOf(checker);
+                break;
+            }
+        }
+
+        // if no open milestone, set current as last milestone
+        if (currentIndex == -1 && allMilestones.size() > 0) {
+            currentIndex = allMilestones.size() - 1;
+        }
+
+        String alias = contents;
+
+        if (alias.equalsIgnoreCase("current") ||
+                alias.equalsIgnoreCase("curr")) {
+            contents = allMilestones.get(currentIndex).getTitle();
+        } else if (alias.equalsIgnoreCase("previous") ||
+                alias.equalsIgnoreCase("prev")) {
+            if (currentIndex - 1 >= 0) {
+                contents = allMilestones.get(currentIndex - 1).getTitle();
+            }
+        } else if (alias.equalsIgnoreCase("next")) {
+            if (currentIndex + 1 < allMilestones.size()) {
+                contents = allMilestones.get(currentIndex + 1).getTitle();
+            }
+        } else if (alias.startsWith("previous-") ||
+                alias.startsWith("prev-")) {
+            int offsetPos = alias.indexOf('-') + 1;
+            int offset = Integer.parseInt(alias.substring(offsetPos).trim());
+            if (currentIndex - 1 - offset >= 0) {
+                contents = allMilestones.get(currentIndex - 1 - offset).getTitle();
+            }
+        } else if (alias.startsWith("next+")) {
+            int offsetPos = alias.indexOf('+') + 1;
+            int offset = Integer.parseInt(alias.substring(offsetPos).trim());
+            if (currentIndex + 1 + offset < allMilestones.size()) {
+                contents = allMilestones.get(currentIndex + 1 + offset).getTitle();
+            }
+        }
+
+        contents = contents.toLowerCase();
+
```

Comment:
You should probably use a regex for this. It would be a lot easier to reason about than all the explicit string processing.

Also, this method will be invoked for every issue, so we probably shouldn't sort over and over in here. One alternative I can think of is to preprocess the filter expression and replace the alias with its concrete equivalent, so it's only done once.